python/ray/tune/syncer.py

```diff
@@ -131,6 +138,13 @@ def __post_init__(self):
             )
         if not self.upload_dir and isinstance(self.syncer, Syncer):
             raise ValueError("Must specify an `upload_dir` to use a custom `syncer`.")
+        if not self.upload_dir:
+            # Can't upload artifacts without specifying upload_dir
+            self.sync_artifacts = False
+
+        if self.upload_dir and self.syncer == "auto":
+            # Setup the default cloud syncer
+            self.syncer = get_node_to_storage_syncer(self)
```

Comment:
I see, I was thinking that this is fine because syncer="auto" is pretty much never manually specified by the user, and it would be nice to just be working with `None` or a `Syncer` object in `sync_config.syncer`.

I am happy to change this back though, and just do this `get_node_to_storage_syncer` elsewhere.

Reply:
+1. nicer to keep the config intact, especially if we save these configs for restore, in which case the behaviors will always be the same.